packages/react/src/primitives/types/collection.ts

```diff
@@ -60,38 +60,52 @@ export interface CollectionWrapperProps extends BaseStyleProps {
   searchNoResultsFound?: React.ReactNode;
 }
 
-export interface CollectionBaseProps<Item> {
+interface CollectionChildren<Item> {
   /**
    * @description
-   * The items from a data source that will be mapped by the Collection component
+   * The component to be repeated
+   * Same interface as Array.prototype.map
    */
-  items: Array<Item>;
+  children: (item: Item, index: number) => JSX.Element;
+}
 
+export interface CollectionBaseProps<Item> extends CollectionChildren<Item> {
   /**
    * @description
-   * The component to be repeated
-   * Same interface as Array.prototype.map
+   * The items from a data source that will be mapped by the Collection component
    */
-  children: (item: Item, index: number) => JSX.Element;
+  items: Array<Item>;
 }
 
-// Omit `children` prop of `BaseFlexProps` and `BaseGridProps` to prevent `CollectionProps[`children']`
-// from resolving to a type union of `React.ReactNode & (item: Item, index: number) => JSX.Element`
-export type ListCollectionProps<Item> = Omit<BaseFlexProps, 'children'> &
+export type ListCollectionProps<Item> = BaseFlexProps &
   CollectionBaseProps<Item>;
-export type GridCollectionProps<Item> = Omit<BaseGridProps, 'children'> &
+export type GridCollectionProps<Item> = BaseGridProps &
   CollectionBaseProps<Item>;
 
+/**
+ * Omits `React.ReactNode` as children to prevent intersection type for `children` of
+ * `React.ReactNode & (item: Item, index: number) => JSX.Element`
+ * and replaces with `CollectionChildren`
+ */
+type ReplaceChildren<T, Item> = Omit<T, 'children'> & CollectionChildren<Item>;
+
+// Note: `BaseCollectionProps` is used directly as the expected props interface of `Collection`
 export type BaseCollectionProps<
   Item,
   Element extends ElementType
 > = PrimitivePropsWithAs<CollectionWrapperProps, Element> &
   (
-    | ({ type: 'list' } & ListCollectionProps<Item>)
-    | ({ type: 'grid' } & GridCollectionProps<Item>)
+    | ReplaceChildren<{ type: 'list' } & ListCollectionProps<Item>, Item>
+    | ReplaceChildren<{ type: 'grid' } & GridCollectionProps<Item>, Item>
   );
 
 export type CollectionProps<
   Item,
   Element extends ElementType = 'div'
-> = PrimitiveProps<BaseCollectionProps<Item, Element>, Element>;
+> = ReplaceChildren<
+  PrimitiveProps<
+    BaseCollectionProps<Item, Element> & { children: React.ReactNode },
```

Comment:
Think this can be replaced by `React.PropsWithChildren<BaseCollectionProps<Item, Element>>`?

Reply:
Think so but unless a blocker would prefer not to make any further changes